polkadot/transaction-pool/src/lib.rs

```diff
@@ -44,19 +47,66 @@ pub fn truncate_id(id: &AccountId) -> TruncatedAccountId {
 	TruncatedAccountId::from_slice(&id[..20])
 }
 
+/// Useful functions for working with Polkadot blocks.
+pub struct PolkadotBlock (Block);
+
+impl PolkadotBlock {
+	/// Create a new block, checking high-level well-formedness.
+	pub fn from(unchecked: Block) -> ::std::result::Result<Self, Block> {
+		if unchecked.extrinsics.len() < 1 {
+			return Err(unchecked);
+		}
+		if unchecked.extrinsics[0].is_signed() {
+			return Err(unchecked);
+		}
+		if let Call::Timestamp(TimestampCall::set(_)) = unchecked.extrinsics[0].extrinsic.function {} else {
```

Comment:
could rewrite this as 

```rust
if let Call::Timestamp(...) = ... { 
    Ok(...) 
} else { 
    Err(...)
}
```

Reply:
would work right now, but not in general since we'd expect more validity tests to be added here.

Reply:
that's fine. the `if let X {} else {}` is unidiomatic so I'd prefer a match statement.